tests/test_unreachable.py

```diff
@@ -0,0 +1,84 @@
+from . import v
+assert v  # Silence pyflakes
+
+
+def check_unreachable(v, lineno, size, name):
+    assert len(v.unreachable_code) == 1
+    item = v.unreachable_code[0]
+    assert item.lineno == lineno
+    assert item.size == size
+    assert item.name == name
+
+
+def test_assignment(v):
+    v.scan("""\
+def foo():
+    print("Hello World")
+    return
+    a = 1
+""")
+    check_unreachable(v, 4, 1, 'return')
+
+
+def test_multiline_return_statements(v):
+    v.scan("""\
+def foo():
+    print("Something")
+    return (something,
+            that,
+            spans,
+            over,
+            multiple,
+            lines)
+    print("Hello World")
+""")
+    check_unreachable(v, 9, 1, 'return')
+
+
+def test_multiple_return_statements(v):
+    v.scan("""\
+def foo():
+    return something
+    return None
+    return (some, statement)
+""")
+    check_unreachable(v, 3, 2, 'return')
+
+
+def test_pass(v):
+    v.scan("""\
+def foo():
+    return
+    pass
+    return something
+""")
+    check_unreachable(v, 3, 2, 'return')
+
+
+def test_recursive_functions(v):
+    v.scan("""\
+def foo(a):
+    if a==1:
+        return 1
+    else:
+        return foo(a-1)
```

Comment:
even though this is text, it should follow our coding conventions: `a == 1` and `a - 1`.